rpc_proxy.go

```diff
@@ -549,6 +565,56 @@ func (p *rpcProxy) basicAuthToMacaroon(basicAuth, requestURI string,
 	}
 }
 
+// convertSuperMacaroon converts a super macaroon into a daemon specific
+// macaroon, but only if the super macaroon contains all required permissions
+// and the target daemon is actually running in remote mode.
+func (p *rpcProxy) convertSuperMacaroon(ctx context.Context, macHex string,
+	fullMethod string) ([]byte, error) {
+
+	requiredPermissions, ok := p.permissionMap[fullMethod]
+	if !ok {
+		return nil, fmt.Errorf("%s: unknown permissions required for "+
+			"method", fullMethod)
+	}
+
+	// We have a super macaroon, from here on out we'll return errors if
+	// something isn't the way we expect it to be.
+	macBytes, err := hex.DecodeString(macHex)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure the super macaroon is valid and contains all the required
+	// permissions.
+	err = p.superMacValidator(
+		ctx, macBytes, requiredPermissions, fullMethod,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Is this actually a request that goes to a daemon that is running
+	// remotely?
+	switch {
+	case isFaradayURI(fullMethod) && p.cfg.faradayRemote:
+		return readMacaroon(lncfg.CleanAndExpandPath(
+			p.cfg.Remote.Faraday.MacaroonPath,
+		))
+
+	case isLoopURI(fullMethod) && p.cfg.loopRemote:
+		return readMacaroon(lncfg.CleanAndExpandPath(
+			p.cfg.Remote.Loop.MacaroonPath,
```

Comment:
Hmm, good question. But I don't think this should cause any issues. Because as you correctly suspect, the user won't see the substituted macaroon on their call. So even if we upgrade from a super macaroon that only has sufficient perms for the current call, they couldn't re-use the substituted macaroon for another call.
But I'm going to add a comment to https://github.com/lightninglabs/lightning-terminal/issues/321 to mention we should add a test that does some basic read/write permission checks.